### PR TITLE
Add dtype to np.cross calls in box_geometry

### PR DESCRIPTION
**Pull request by ccaven**
Add dtype flag to `np.cross` calls to stop Pylance from parsing as `NoReturn`

Changed 
- `pygfx/geometries/_box.py`
- `pygfx/geometries/_polyhedron.py` 
- `pygfx/geometries/_toroidal.py`

#389

### CODE REVIEW

**Review by Korijn — APPROVED** [review]

**Comment by Korijn** [comment]
I also find this annoying enough to merge this. I think newer versions of numpy will include the fix.

**Comment by ccaven** [comment]
> I also find this annoying enough to merge this. I think newer versions of numpy will include the fix.

btw, I wrongly assumed `cross` accepts a `dtype` parameter. It doesn't, so to fix the issue I used `np.ndarray.astype` before passing arrays into the `cross` function, e.g.
image

File: pygfx/geometries/_box.py [hunk 1]
```diff
@@ -52,12 +52,16 @@ def box_geometry(
     plane_csys = np.array(
         [
             *cube_normal_up[4],
-            np.cross(*cube_normal_up[4]),
+            np.cross(
+                np.ndarray.astype(cube_normal_up[4, 0], np.float32),
+                np.ndarray.astype(cube_normal_up[4, 1], np.float32)
```
Comment (Korijn) [inline pygfx/geometries/_box.py:57]: Can you pass `copy=False` as well? That way no copy will be made if the input is already float32

Comment (ccaven) [reply pygfx/geometries/_box.py:57]: Yes - will do when I get back home

Comment (ccaven) [reply pygfx/geometries/_box.py]: Done

**Comment by ccaven** [comment]
Sorry, I didn't check the linting requirements. Since I had to fix those anyways, I moved the changes in `_box.py` upsteam:
image

**Comment by ccaven** [comment]
apparently this linter is out to get me lol
removed the whitespace

**Comment by ccaven** [comment]
image

i have no clue the exact rules for this so i just installed `flake8-black`  and used it to format the file lol. hopefully it accepts this time

**Review by Korijn — APPROVED** [review]

**Comment by Korijn** [comment]
Thank you for the contribution! :)

**Pull request merged**